packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -248,21 +236,51 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
   }
 
   // ====== RestAPI ======
-  RestOperation _restFunctionHelper(
-      {required String methodName, required RestOptions restOptions}) {
-    // Send Request cancelToken to Native
-    String cancelToken = UUID.getUUID();
 
-    Future<RestResponse> futureResponse =
-        _callNativeRestMethod(methodName, cancelToken, restOptions);
+  Future<AWSStreamedHttpResponse> _restResponseHelper({
+    required String methodName,
+    required String path,
+    required String cancelToken,
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, dynamic>? queryParameters,
+    String? apiName,
+  }) async {
+    final bodyBytes =
+        body != null ? Uint8List.fromList(await body.single) : null;
```

Comment:
You'll need to fold the bytes since there could be multiple chunks (see AWSStreamedHttpResponse.bodyBytes). Imagine if `body` is `Stream.fromIterable([[1,2,3], [4,5,6]])`, this would result in `Uint8List.fromList([1,2,3])` which is incorrect.

Reply:
thanks, took a shot at fixing in next revision, lmk if needs more changes.